code/modules/atmospherics/gasmixtures/reactions.dm

```diff
@@ -0,0 +1,209 @@
+#define NO_REACTION	0
+#define REACTING	1
+
+var/list/gas_reactions = init_gas_reactions() //this is our singleton of all reactions
+/proc/init_gas_reactions()
+	. = new /list
+	var/list/reaction_types = typesof(/datum/gas_reaction - /datum/gas_reaction)
+
+	for(var/i in 2 to reaction_types.len) //holy shit there's gotta be a better way of sorting this lmao. sorts in descending order of priority
+		var/j = i
+		var/datum/gas_reaction/a = reaction_types[j-1]
+		var/datum/gas_reaction/b = reaction_types[j]
+		while(j > 0 && initial(a.priority) < initial(b.priority))
+			reaction_types.Swap(j, j-1)
+			j--
+			a = reaction_types[j-1]
+			b = reaction_types[j]
+
+	for(var/path in reaction_types)
+		. += new path
+
+/datum/gas_reaction
+	var/min_requirements
+	var/max_requirements
+	var/exclude = FALSE //do it this way to allow for addition/removal of reactions midmatch in the future
+	var/priority //lower numbers are checked/react later than higher numbers. if two reactions have the same priority they may happen in either order
+
+/datum/gas_reaction/New()
+	init_reqs()
+
+/datum/gas_reaction/proc/init_reqs()
+/datum/gas_reaction/proc/react(datum/gas_mixture/air, atom/location)
+	return NO_REACTION
+
+//agent b: converts hot co2 and agent b to oxygen. requires plasma as a catalyst. endothermic
+/datum/gas_reaction/agent_b
+	priority = 2
+
+/datum/gas_reaction/agent_b/init_reqs()
+	min_requirements = "TEMP=900;agent_b=0;plasma=[MINIMUM_HEAT_CAPACITY];co2=[MINIMUM_HEAT_CAPACITY]"
+
+/datum/gas_reaction/agent_b/react(datum/gas_mixture/air)
+	var/list/cached_gases = air.gases
+	var/reaction_rate = min(cached_gases["co2"][MOLES]*0.75, cached_gases["plasma"][MOLES]*0.25, cached_gases["agent_b"][MOLES]*0.05)
+
+	cached_gases["co2"][MOLES] -= reaction_rate
+	cached_gases["agent_b"][MOLES] -= reaction_rate*0.05
+
+	air.assert_gas("o2") //only need to assert oxygen, as this reaction doesn't occur without the other gases existing
+	cached_gases["o2"][MOLES] += reaction_rate
+
+	air.temperature -= (reaction_rate*20000)/air.heat_capacity()
+
+	return REACTING
+
+//freon: does a freezy thing?
+/datum/gas_reaction/freon
+	priority = 1
+
+/datum/gas_reaction/freon/init_reqs()
+	min_requirements = "freon=[MOLES_PLASMA_VISIBLE]"
+
+/datum/gas_reaction/freon/react(datum/gas_mixture/air, turf/open/location)
+	. = NO_REACTION
+	if(location.freon_gas_act())
+		air.gases["freon"][MOLES] -= MOLES_PLASMA_VISIBLE
```

Comment:
nah, it can be removed in a later PR but as always I want to maintain functionality as much as possible